src/timepicker/timepicker.js

```diff
@@ -200,24 +201,45 @@ class TimePicker<T = Date> extends React.Component<
   };
 
   getTimeWindowInSeconds = (step: number): {start: number, end: number} => {
-    let {minTime: min, maxTime: max} = this.props;
-    let midnight = this.setTime(this.props.value, 0, 0, 0);
-    if (!min) {
-      min = midnight;
+    let {minTime: min, maxTime: max, ignoreMinMaxDateComponent} = this.props;
+    const dayStart = this.setTime(this.props.value, 0, 0, 0);
+    const dayEnd = this.setTime(this.props.value, 24, 0, 0);
+
+    if (
+      !min ||
+      (this.props.adapter.toJsDate(min) <
+        this.props.adapter.toJsDate(dayStart) &&
```

Comment:
small nit: can we use `this.props.adapter.isBefore` & `this.props.adapter.isAfter` instead of less than or greater than, I think the code will be more readable then.
```suggestion
      (this.props.adapter.isBefore(min, dayStart) &&
```

Reply:
Thank you